components/org.wso2.carbon.identity.application.authenticator.push/src/main/java/org/wso2/carbon/identity/application/authenticator/push/PushAuthenticator.java

```diff
@@ -208,8 +232,14 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
             throw new AuthenticationFailedException(errorMessage);
         }
 
-        contextManager.clearContext(getClaimFromClaimSet(claimsSet,
-                PushAuthenticatorConstants.TOKEN_SESSION_DATA_KEY, deviceId));
+        try {
+            contextManager.clearContext(validator.getClaimFromClaimSet(claimsSet,
```

Comment:
I see that we are retrieving the `TOKEN_SESSION_DATA_KEY` in a previous step in this method. can we use it here without again retrieving it from the claimset?

Reply:
It was a mistake. therefore the session data key is not retrieved earlier